src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/VmrManagerBase.cs

```diff
@@ -271,12 +278,17 @@ protected async Task<IEnumerable<VmrDependencyUpdate>> GetAllDependenciesAsync(
                         $"for a {VersionFiles.VersionDetailsXml} dependency of {dependency.Name}");
                 }
 
+                var build = (await barClient.GetBuildsAsync(dependency.RepoUri, dependency.RepoUri))
+                        .SingleOrDefault();
```

Comment:
I wonder if `FirstOrDefault` will be better than nothing. Thoughts?

Reply:
`FirstOrDefault` could be incorrect tho, in cases where we have multiple builds from the same commit. I'm not sure if we should try to get the newest build or do something like that tho. Currently, if this does happen, we'll still do the code flow, but won't have the OfficialBuildId, or the BarId. This could be a topic for the meeting on tuesday

Reply:
I somehow still think it's better to have an approximately good information than nothing since the `OfficialBuildId` is required there for SB purposes.

Reply:
+1 for logging